sipp/src/lib.rs

```diff
@@ -128,18 +155,16 @@ impl<E: PairingEngine, D: Digest> SIPP<E, D> {
         ark_ff::batch_inversion(&mut x_invs);
 
         let z_prime = claimed_value
-            * &proof
+            + proof
                 .gt_elems
                 .par_iter()
                 .zip(&x_s)
                 .zip(&x_invs)
-                .map(|(((z_l, z_r), x), x_inv)| {
-                    z_l.pow(x.into_repr()) * &z_r.pow(x_inv.into_repr())
-                })
-                .reduce(|| E::Fqk::one(), |a, b| a * &b);
+                .map(|(((z_l, z_r), x), x_inv)| (*z_l * x) + (*z_r * x_inv))
+                .reduce(|| PairingOutput::<E>::default(), |a, b| a + b);
```

Comment:
The bug is probably coming from here

```suggestion
                .reduce(|| PairingOutput::<E>::zero(), |a, b| a + b);
```

Reply:
Good catch. Actually this is pretty bad, because the default value of this type (0) is not even in the subgroup. I'll make a change in `algebra` to either remove it or make it 1 (I really doubt anyone is currently relying on the fact that it's 0, since you can't do anything with it when it's 0)

Reply:
Yeah, changing the `Default::default` impl to output `1` makes sense.